### ISSUE TITLE
Sending amount of 0 sends all bitcoins

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#637]
If I supply a value of 0 to the api endpoint for sending balance, it sends my WHOLE balance to the endpoint. 

I understand that we have undocumented functionality where if you don't supply the `amount` argument, that the entire balance will be sent (we use this functionality to send the whole balance from the core wallet to the browser wallet). However, I would expect that if I supply an amount that that amount would be used. Supplying 0 or some value that's too small to be used to make a transaction should be a `400` error. 

This is a pretty big bug - someone could easily inadvertently send their entire wallet balance to the wrong person because of this. I'll add some additional error checking on the browser UI, but we should also make sure that the API currently interprets 0 as 0 instead as "send all" as well.

```
[2017-10-07 22:37:43,304] [DEBUG] [rpc:3917] (31808.123145531961344) 
full path: /v1/wallet/balance
method: POST
path: /v1/wallet/balance
qs: {}
headers:
origin: http://localhost:8888
content-length: 145
accept-language: en-US,en;q=0.8
accept-encoding: gzip, deflate, br
connection: keep-alive
accept: application/json
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36
host: localhost:6270
referer: http://localhost:8888/wallet/send
content-type: application/json
authorization: bearer D247390B-8F11-4D02-966B-992FECFFD652-40569-000106AB0F3AB03C

[2017-10-07 22:37:43,417] [DEBUG] [app:152] (31808.123145531961344) Not a valid token
[2017-10-07 22:37:43,529] [DEBUG] [rpc:3964] (31808.123145531961344) Authenticated with password
[2017-10-07 22:37:43,815] [WARNING] [blockchain:134] (31808.123145531961344) Using a different number of confirmations (0) instead of default (6)
[2017-10-07 22:37:43,931] [WARNING] [scripts:383] (31808.123145531961344) Using UTXOs with 0 confirmations instead of the default 6
[2017-10-07 22:37:44,040] [DEBUG] [insight_api:41] (31797.123145531961344) GET https://utxo.blockstack.org/insight-api/addr/1ocXpu8pTzHEq4KQkgYNe7W6fwzAvyYT8/utxo
[2017-10-07 22:37:45,009] [DEBUG] [insight_api:52] (31797.123145531961344) 1ocXpu8pTzHEq4KQkgYNe7W6fwzAvyYT8 has 2 UTXOs
[2017-10-07 22:37:45,010] [DEBUG] [actions:535] (31808.123145531961344) No amount given; assuming all (349199)
[2017-10-07 22:37:47,219] [DEBUG] [session:189] (31797.123145531961344) Connect to bitcoind at bitcoin.blockstack.com:8332 (/Users/larry/Library/Application Support/Blockstack/config/client.ini)
[2017-10-07 22:37:47,220] [DEBUG] [session:104] (31797.123145531961344) [31808] Connect to bitcoind at http://blockstack@bitcoin.blockstack.com:8332, timeout=300.0
[2017-10-07 22:37:47,824] [DEBUG] [fees:87] (31797.123145531961344) Bitcoin estimatefee(2) is 0.00149672
[2017-10-07 22:37:47,825] [DEBUG] [fees:93] (31797.123145531961344) Bitcoin estimatefee(2) is 0.00149672 (146 satoshi/byte)
[2017-10-07 22:37:47,825] [DEBUG] [actions:535] (31808.123145531961344) No amount given; assuming all (299705)
[2017-10-07 22:37:50,007] [DEBUG] [ecdsalib:240] (31797.123145531961344) High-S to low-S
[2017-10-07 22:37:50,008] [DEBUG] [actions:597] (31808.123145531961344) Withdraw None from 1ocXpu8pTzHEq4KQkgYNe7W6fwzAvyYT8 to 15e21SDn6pzdnh4SMExGufbfSepWMik8Cr
[2017-10-07 22:37:50,120] [DEBUG] [blockchain:217] (31808.123145531961344) Send 339-byte tx 0100000002ea3fe4c70cea3abac3d6c8239226b1ecdecd93ffd78e2fde9f3815a75b6e82dc000000006b483045022100b3decf3fbd36c52268926af63f0866a71cecb802b08bca6db1b538ba8c4450680220796c04a5de91bf29855ccdc651e10f7859564afa958ed0185a7f64ddc8a8affa01210392ec4bba8394d24e88716e347345bbc7440f02698f7ea6add58b007a34f218a6feffffffdc93b3c1bb98bcb406b3a1489d01d7fd9b6b744e3cdf634b44d1d8183d832229020000006a47304402203f94fc52a2173c0db8a006b4773e3da15ed8afbce1f60b82fcab665163e2dde202201b5d89683f1698db98f432d5723350762d6c61b30adcf60543959e648da2b77001210392ec4bba8394d24e88716e347345bbc7440f02698f7ea6add58b007a34f218a6feffffff01b9920400000000001976a91432e0bc17c2a5018c6391b0331ef56f93b25302c088ac00000000
[2017-10-07 22:37:50,229] [DEBUG] [insight_api:69] (31797.123145531961344) POST https://utxo.blockstack.org/insight-api/tx/send
127.0.0.1 - - [07/Oct/2017 22:37:51] "POST /v1/wallet/balance HTTP/1.1" 200 -
```

**Comment by kantai** [comment]
This is fixed with PR #638 -- 

However, the browser I was using to test (current `develop` head: 082c0d21436f689863e15ea395725ba47ca92836) wasn't sending the 'amount' parameter when I set it to 0:

The post data:
```
{"address":"16EMaNw3pkn3v6f2BgnSSs53zAKH4Q8YJg","min_confs":0,"payment_key":"f90d6049cdf55a32a2f8e869dbc581bafc57091784b6ba223f36d1a95a70ef1201"}
```

**Comment by larrysalibra** [comment]
You're right, browser wasn't sending the `amount` parameter. This is fixed in browser as of https://github.com/blockstack/blockstack-browser/commit/1b4183a90f7dd1edbb5aa67e37ee9e0d15e091aa

Thanks for your help!
